{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Week1</h6>\n",
    "<p>Experimental Design Varibility, Replication and Power</p>\n",
    "<p>Central dogma of biological statistics we have a population, we pick a sample and take some measurements from \n",
    "that sample and we want to infer properties of the population. We can't sample the entire population\n",
    "because of cost. </p>\n",
    "<img src=\"week1/week1_1.png\">\n",
    "<p>3 source of variability</p>\n",
    "<li>Phenotypic variability, range of phenotype, variability in phenotype measurement. ie. output of crop</li>\n",
    "<li>Measurement error, big area for attention, 2 types random measurement error or correlated/biased batch effects cover later</li>\n",
    "<li>biological variaton,any 2 people will have difference in genomics. Variation in amount of each gene expressed\n",
    "or number of types of cells in body</li>\n",
    "<img src=\"week1/week1_2.png\">\n",
    "<p>how to measure variability as you replicate experiment? 2 types of replicates, technical replicates and biological \n",
    "replicates. Tecnical Replicate:do same experiment twice using the same biological sample, ie rna sequencing and rna microarray\n",
    "    Biological Replicate:take different people or different organisms and measure same gene.</p>\n",
    "<img src=\"week1/week1_3.png\">\n",
    "<p>sample sizes depends on bio experiment, N is biological replicates.</p>\n",
    "<img src=\"week1/week1_4.png\">\n",
    "<p>One issue is biological variability, Need N large enough to account for biological variability as\n",
    "shown by the range of gene expresssion in red and blue. Another example is making sure dosing is high \n",
    "enough to affect the population based on biological variation.</p>\n",
    "<img src=\"week1/week1_5.png\">\n",
    "<p>Statistical power is the probability of the hypothesis test detecting an effect; if there is one. This \n",
    "analysis can be used to find the minimum sample size N. The null hypothesis test makes an assumption\n",
    "about the outcome. The output of the test is a p-valu which is the probability of observing\n",
    "the result if the null hypothesis is true. You pick a significance level to determing statistical significane.\n",
    "If the result p-value is less than the statistical significane p-value then teh null hyphthesis is accepted. if the \n",
    "calculated p-value is greather than the statistical significance p-value then teh nuyll hypothesis is rejected. </p>\n",
    "<p>The 4 related parameters are 1) Effect Size,the result, ie pearsons correlation coefficient for the relationship\n",
    "between 2 variables. 2) Sample Size,N 3) Significance, p value to define statistical significance\n",
    "usually set to .05 or 5%. 4) Statistical power, usually set\n",
    "at 80%. These 4 variables are all related. A larger N means the effect can be easier to detect(this is true? how about\n",
    "bonferroni adjusting the p by p/N? statistical power can be increased by decreasing the signifcance level. </p>\n",
    "<p>Example:Student t-test for comparing if the means of 2 samples are identical even though the mean is not \n",
    "numerically identical. ie. is a mean of 3.4 the same as mean of 4? p_alpha=0.05, effect_size=.80, power = 0.8.\n",
    "With these defaults what the number of observations from each sample/batch to detect an effect of 0.8 with \n",
    "an 80% chance of detecting the effect if true? 20% for type2 error, 5% if there is no such effect or type 1 error. \n",
    "False negative rate is type 2 error? Yes, type 1 error is false positive. </p>\n",
    "                                                                                               \n",
    "<img src=\"week1/week1_6.png\">\n",
    "<p></p>\n",
    "<img src=\"week1/week1_7.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "<h6>week1 confounding and randomization</h6>\n",
    "<p>The solution to the confounding problem is randomization</p>\n",
    "<p>Confounding is when you are looking at the relationship between 2 variables and there is another \n",
    "hidden variable affecting the output. This hidden variable can affect results. \n",
    "If we are looking for an association between gene expression levels and\n",
    "control/disease group or phenotype there may be other variables affecting this relationship. Another var\n",
    "is a confounding variable. Example kid and adult there is an output, literate, adult is literate kid is not. And\n",
    "you can correlate show size, kid small, adult large so does large shoe size mean literate? </p>\n",
    "<img src=\"week1/week1_2a.png\">\n",
    "<p>Confounding example: there is an intermediate variable age which would model the relationship of literacy. The confounding variable is related to \n",
    "both age and literacy. JL compares a literate adult to an illiterate child. The input can be shoe size and that\n",
    "is correlated to litteracy levels. But the hidden variable is really number of years of education or age assuming age\n",
    "correlates with years of education</p>\n",
    "<img src=\"week1/week1_2b.png\">\n",
    "<p>GSEA example on confounding variable problem: 78% differentially expressed when you look at asian vs eur but \n",
    "    96% diff. expresseed when you look at year. So year was a more important predictor. And this study was fishy\n",
    "    because past studies showed no correlations between ethnicity and expression. Turns out the data was collected\n",
    "    using different methods, ie microarray vs. sequencing and so variations were due to technology variation\n",
    "    not from the ethnicity or time variable! </p>\n",
    "<img src=\"week1/week1_2c.png\">\n",
    "<p>note the clustering on years, time differences from technology variation not genetic variation! </p>\n",
    "<img src=\"week1/week1_2d.png\">\n",
    "<p>When you adjust for years you see no statistically significant effect! How do you do the correction?</p>\n",
    "<img src=\"week1/week1_2e.png\">\n",
    "<p>study showed people >100 had genetic variants that caused longevity. turns out this was an error \n",
    "in technology variation bc samples on young people were collected on 1 technology and the samples on \n",
    "older were collected on another tech. </p>\n",
    "<img src=\"week1/week1_2f.png\">\n",
    "<p>occurs in proteomics; correlated a protemic variation to cancer but turns out there was a batch effect\n",
    "related to time so you were measuring time variation vs. biology variation</p>\n",
    "<img src=\"week1/week1_2g.png\">\n",
    "<p>paper worth reading talks about batch effects. Batch effects include operator, reagent biases, high throughput\n",
    "systems can remove these, low throughput cant. Normaalization is common for GSEA,and is only 1 technique. \n",
    "ie normalization across microarrray and \n",
    "sequender output. Normalization reduces variance w/o changing bias if done correctly. Normalization has 2 components\n",
    "1) scale intensities so each method/batch has same value. 2) transorm data using (x-mu)/sigma so distribution of probe intensities\n",
    "is same on all samples is the simplistic view. Not what is done in practice. \n",
    "</p>\n",
    "Batch effects is complicated field, open source sw ranging from PCA to canonical correlation analysis, etc.. many techniques\n",
    "one simplistic view is a tsne plot of batch corrected data. Looks better in that visually batch effect isnt adding \n",
    "confounding variables to the data. But still needs some work to make into system heuristics for quality. \n",
    "<img src=\"week1/week1_2h.png\">\n",
    "<img src=\"week1/week1_2i.png\">\n",
    "<img src=\"week1/week1_2j.png\">\n",
    "<h6>Adding randomization to confounding variable</h6>\n",
    "<p>Things you can do at the experiment design stage for batch effects. \n",
    "dark means confounding variable is high, light means low. When you do the experiment, the red circles are \n",
    "the samples in 1 treatment batch and the green another treatment batch so you have measured the confounding variable and not the \n",
    "real gene expression. </p>\n",
    "<img src=\"week1/week1_2k.png\">\n",
    "<p>another way to design the experiment for each patient you randomly assign a green or red treatment.</p>\n",
    "<img src=\"week1/week1_2l.png\">\n",
    "<p>Another design method: blocking</p>\n",
    "<img src=\"week1/week1_2m.png\">\n",
    "<p>If run all female Control in week 1 and all male Treated in week2 hard to distinguish between \n",
    "male/femake effect, week1, week2 effect, control/disease effect. </p>\n",
    "<img src=\"week1/week1_2n.png\">\n",
    "<p>block the subjects to get equal balance</p>\n",
    "<img src=\"week1/week1_2o.png\">\n",
    "<p>control, balanced means same yellow/green in each slot. Lot more info on experiment design, balanced replicatin, controls.\n",
    "Q: how do you know from data if somethign is balanced? Uneven variances, uneven N. </p>\n",
    "<img src=\"week1/week1_2p.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "<h6>data exploration</h6 >\n",
    "<img src=\"week1/week1_3a.png\">\n",
    "visually comparing 2 dots side by side is easier to see the difference in values,in contrast to \n",
    "a pie chart. This is not GSEA specific. \n",
    "<img src=\"week1/week1_3b.png\">\n",
    "<img src=\"week1/week1_3c.png\">\n",
    "<img src=\"week1/week1_3d.png\">\n",
    "<img src=\"week1/week1_3e.png\">\n",
    "<img src=\"week1/week1_3f.png\">\n",
    "<img src=\"week1/week1_3g.png\">\n",
    "<p>Bland-Altman plot or Tukey plot,visualize 2 assays on same sample.\n",
    "take difference between y values. </p>\n",
    "<img src=\"week1/week1_3h.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
